Improve OpenTelemetry Java manual instrumentation documentation (#19424)
DataDog/documentation: content/en/tracing/trace_collection/otel_instrumentation/java.md

The following OpenTelemetry features implemented in the Datadog library as noted:

> 

Reviewer: I'd suggest noting that for those listed as "Unsupported", there may be progress on moving those to "Supported" in the near future, and if a customer wants one of those features, they should escalate to support about this.

Author: Agreed. I just add the instrumentation annotation for example.
But the contributing guideline does not recommend such content ([ref](https://github.com/DataDog/documentation/blob/master/CONTRIBUTING.md#tense)):

> Avoid temporal words like "currently", "now", "will", etc. Describe the present state of the product.

So, what would be the right sentence to add below this table?

> The support of OpenTelemetry features is an ongoing effort. If there is any missing feature of your interest, please escalate to support about it.

|---------------------------------------------------|--------------------------------------------------------------------|
| [OpenTelemetry Context propagation][1]            | [Datadog distributed header format][9] is used instead by default. | 
| [Span processors][2]                              | Unsupported                                                        | 
| [Span Exporters][3]                               | Unsupported                                                        |
| Trace/span [ID generators][4]                     | ID generation is performed by `ddtrace`.                           |
| [Metrics][7], [Baggage][12] and [Context][10] API | Unsupported                                                        |
| Instrumentation [annotations][6]                  | Unsupported                                                        |

## Configuring OpenTelemetry to use the Datadog tracing library

1. Add your desired manual OpenTelemetry instrumentation to your Java code following the [OpenTelemetry Java Manual Instrumentation documentation][5].

1. Add the [Datadog tracing library to the JVM][11].

> 1. Make sure you only depends from OpenTelemetry API (and not OpenTelemetry SDK).

Reviewer 2: ```suggestion
1. Make sure you only depend on the OpenTelemetry API (and not OpenTelemetry SDK).
```
I think this is what you meant to say? ^

Author: Yes, I think so. Thanks!

    Using Maven:
    ```xml
      <!-- OpenTelemetry API -->
      <dependency>
              <groupId>io.opentelemetry</groupId>
              <artifactId>opentelemetry-api</artifactId>
              <version>${io.opentelemtry.version}</version>
          </dependency>
      <dependency>
    ```

    Using Gradle:
    ```groovy
      // OpenTelemetry API
      implementation "io.opentelemetry:opentelemetry-api:${opentelemetryVersion}"
    ```

1. Set the `dd.trace.otel.enabled` system property to `true`.

Datadog combines these OpenTelemetry spans with other Datadog APM spans into a single trace of your application.

> ## Common Use Cases

Reviewer 2: ```suggestion
## Common use cases
```

### Add custom attributes to the current or local root span

```java
// Add attributes to the current span
Span currentSpan = Span.current();
currentSpan.setAttributes("some-key", "some-value");

// Add attributes to the local root span
ContextKey<OtelSpan> localRootSpanKey = ContextKey.named("datadog-root-span-key");
Span rootSpan = Context.current().get(localRootSpanKey);
rootSpan.setAttributes("some-key", "some-value");
```

> **Note:** When no current or local root span, the returned span will not be `null` but invalid, and attributes will not be set.

Reviewer 2: ```suggestion
**Note:** If there isn't a current or local root span, the returned span is invalid, not `null`, and attributes are not set.
```
Addressing Vale check ^

Author: That't way better! Thank you 🙏
